### PR TITLE
#161596122 Delete document

### PR DESCRIPTION
**Pull request by Nedson202**
#### What does this PR do?
Handles deletion of a users' document

#### Description of Task to be completed?
- add tests for deletion method and endpoint
- add the method to delete a users' document from the database

#### How should this be manually tested?
- Clone the repo - git clone https://github.com/andela/travel_tool_back.git
- Checkout to this branch - git checkout ch-fix-trip-planner-page-refresh-161699088
- Install dependencies and start the server - make start
- Setup the frontend - git clone https://github.com/andela/travel_tool_front.git && make start
- Login on the browser and grab your token from `cookies` in the application tab on chrome dev tools
- Open postman and add the token to the headers tab
- Send a delete request to - http://localhost:5000/api/v1/documents/:documentId

#### Any background context you want to provide?
Check the database for seeded documents and grab an Id. Also check users table, grab your userId and replace it with the userId on the document table.

#### What are the relevant pivotal tracker stories?
[#161596122](https://www.pivotaltracker.com/story/show/161596122)

#### Screenshots (if appropriate)
docus

### CODE REVIEW

**Comment by Spectrumsun** [comment]
update your PR label and fix conflict @Nedson202

**Comment by Nedson202** [comment]
@Spectrumsun the conflicts have been resolved.

File: src/modules/documents/DocumentsController.js [hunk 1]
```diff
@@ -2,8 +2,34 @@ import models from '../../database/models';
 import CustomError from '../../helpers/Error';
 
 const { Op } = models.Sequelize;
-
 export default class DocumentsController {
+  static async deleteDocument(req, res) {
+    const { id } = req.user.UserInfo;
+    const { documentId } = req.params;
+    try {
+      const documentToDelete = await models.Document.findOne({
```
Comment (Spectrumsun) [inline src/modules/documents/DocumentsController.js]: You don't need to use findOne here, You can use destroy directly.

File: src/modules/documents/DocumentsController.js [hunk 2]
```diff
@@ -2,8 +2,34 @@ import models from '../../database/models';
...
+          userId: id
+        }
+      });
+      if (!documentToDelete) {
+        return res.status(404).json({
+          success: false,
+          message: 'Document not found',
+        });
+      }
+      await documentToDelete.destroy();
```
Comment (Spectrumsun) [inline src/modules/documents/DocumentsController.js]: and remove the destroy line from here

**Comment by Nedson202** [comment]
The reason for using findOne is to cater for displaying deleted documents which will probably come in later on. Returning the deletedDocument will help eliminate the need for making queries to the server after a deletion.

**Comment by Nedson202** [comment]
Alright let me try that out.

**Comment by Nedson202** [comment]
Done @Spectrumsun and Thanks.

**Comment by Spectrumsun** [comment]
Good job

**Pull request merged**